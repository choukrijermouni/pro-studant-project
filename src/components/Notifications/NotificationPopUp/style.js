import {
  neutral,
  blue
} from '@pro_boa/ui'
export default {
  notificationPaper: {
    width: 449,
    height: 'auto',
    maxHeight: 603,
    backgroundColor: neutral[0],
    position: 'relative',
    top: 24,
    right: 250,
    display: 'flex',
    flexDirection: 'column'
  },
  notificationCardsContainer: {
    width: ' 100%',
    height: 'calc(100% - 124px)',
    display: 'flex',
    flexDirection: 'column',
    overflow: 'auto',
    marginBottom: 47,
    overflowX: 'hidden',
    '&::-webkit-scrollbar': {
      width: 5
    },
    '&::-webkit-scrollbar-thumb': {
      background: blue[1],
      borderRadius: 50
    }
  },
  notificationContainer: {
    width: 20,
    height: 20
  },
  notificationHeader: {
    height: 78,
    display: 'flex',
    marginTop: '-22px',
    alignItems: 'center',
    justifyContent: 'space-between',
    width: '100%',
    padding: '12px 12px 12px 24px'
  },
  notificationFooter: {
    height: 47,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    width: '100%',
    position: 'absolute',
    backgroundColor: neutral[0],
    bottom: 0,
    padding: 12,
    borderTop: `.3px ${neutral[2]} solid`
  },
  triangle: {
    position: 'relative',
    top: '-17px',
    left: 213
  },
  cursor: {
    cursor: 'pointer'
  }
}
