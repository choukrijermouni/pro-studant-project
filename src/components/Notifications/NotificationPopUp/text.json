{
    "profile": "Compte",
    "name": "Henry Waters",
    "action": " a créé l’équipe",
    "link": " Développement",
    "time": "Il y a 20 min",
    "triangle": "▲",
    "notificationTitle": "Notifications",
    "seen": "Tout marquer comme lu",
    "show": "Afficher toutes les notifications"
}