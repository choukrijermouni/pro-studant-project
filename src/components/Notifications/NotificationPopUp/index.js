import {
  TypographyElement,
  Paper,
  blue,
  neutral
} from '@pro_boa/ui'
import { createUseStyles } from 'react-jss'
import style from './style'
import Text from './text'
import NotificationCard from '../NotificationCard'
import Data from 'mock/notifications'
import { useState } from 'react'
import { push } from 'connected-react-router'
import { NotificationsPath } from 'Routes'
import { useDispatch } from 'react-redux'

const useStyle = createUseStyles(style)

export default () => {
  const dispatch = useDispatch()
  const [unseen, setUnseen] = useState(true)
  const { triangle, notificationHeader, notificationCardsContainer, notificationFooter, notificationPaper, cursor } = useStyle()
  return (
    <Paper className={notificationPaper}>
      <TypographyElement
        component='p'
        variant='heading4'
        align='left'
        fontSize='16px'
        className={triangle}
        lignHeight='15px'
        spacing='-12px 0 0 0'
        display='inline'
        color={neutral[0]}
      >
        {Text.triangle}
      </TypographyElement>
      <div className={notificationHeader}>
        <TypographyElement
          component='h4'
          variant='heading4'
          align='left'
          fontSize='19px'
          lignHeight='24px'
        >
          {Text.notificationTitle}
        </TypographyElement>
        <TypographyElement
          component='p'
          variant='heading4'
          align='left'
          fontSize='12px'
          lignHeight='15px'
          color={blue[0]}
        >
          <div className={cursor} onClick={() => setUnseen(false)}> {Text.seen}</div>
        </TypographyElement>
      </div>
      <div className={notificationCardsContainer}>
        {
        Data.map((Notification, key) => (
          <NotificationCard key={key} Data={Notification} unseen={unseen} />
        ))
      }
      </div>
      <div className={notificationFooter}>
        <TypographyElement
          component='p'
          variant='heading4'
          align='left'
          fontSize='12px'
          lignHeight='15px'
          color={neutral[4]}
        >
          <div className={cursor} onClick={() => dispatch(push(NotificationsPath))}>{Text.show}</div>
        </TypographyElement>
      </div>
    </Paper>
  )
}
