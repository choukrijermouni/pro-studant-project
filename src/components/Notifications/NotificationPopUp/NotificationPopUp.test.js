import { shallow, configure } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import renderer from 'react-test-renderer'
import NotificationPopUp from './'
import { ReduxProvider } from '../../../test/ReduxProvider'

configure({ adapter: new Adapter() })

describe('>>> NotificationPopUp --- Shallow Render NotificationPopUp component', () => {
  let wrapper
  let tree

  beforeEach(() => {
    wrapper = shallow(<ReduxProvider><NotificationPopUp /></ReduxProvider>)
    tree = renderer.create(
      <ReduxProvider><NotificationPopUp /></ReduxProvider>
    ).toJSON()
  })

  it('+++ render correctly and match snapshot', () => {
    expect(tree).toMatchSnapshot()
  })

  it('+++ render the component', () => {
    expect(wrapper).toBeTruthy()
  })
})
