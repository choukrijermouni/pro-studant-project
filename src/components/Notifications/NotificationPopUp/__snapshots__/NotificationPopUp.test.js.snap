// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> NotificationPopUp --- Shallow Render NotificationPopUp component +++ render correctly and match snapshot 1`] = `
<div
  className="notificationPaper-0-2-1 paper-0-2-8"
>
  <p
    className="text-0-2-9 text-d0-0-2-10 triangle-0-2-6"
  >
    ▲
  </p>
  <div
    className="notificationHeader-0-2-4"
  >
    <p
      className="text-0-2-11 text-d0-0-2-12"
    >
      Notifications
    </p>
    <p
      className="text-0-2-13 text-d0-0-2-14"
    >
      <div
        className="cursor-0-2-7"
        onClick={[Function]}
      >
         
        Marquer tout comme lu
      </div>
    </p>
  </div>
  <div
    className="notificationCardsContainer-0-2-2"
  >
    <div
      className="notificationCard-0-2-15 notificationCard-d0-0-2-19"
    >
      <div
        className="avatarContainer-0-2-16"
      >
        <div
          className="avatarContainer-0-2-21"
        >
          <div
            className="dot-0-2-20"
          />
          <img
            className="avatarGroup-0-2-22 avatarGroup-d0-0-2-23"
            src="https://static.pro_boa.com/dev/upload/9f8f4041-acfe-4fcb-99b1-0545a6d35d1e.jpeg"
          />
        </div>
        <div
          className="textContainer-0-2-18"
        >
          <div>
            <p
              className="text-0-2-24 text-d0-0-2-25"
            >
              Choukri Jermouni
            </p>
            <p
              className="text-0-2-26 text-d0-0-2-27"
            >
               à crée l’équipe
            </p>
            <p
              className="text-0-2-28 text-d0-0-2-29"
            >
               Développement
            </p>
          </div>
          <p
            className="text-0-2-30 text-d0-0-2-31"
          >
            Il y a 20 min
          </p>
        </div>
      </div>
      <span
        className="icon-roundedUp icon-0-2-17"
      />
    </div>
    <div
      className="notificationCard-0-2-15 notificationCard-d1-0-2-32"
    >
      <div
        className="avatarContainer-0-2-16"
      >
        <div
          className="avatarContainer-0-2-21"
        >
          <div
            className="dot-0-2-20"
          />
          <img
            className="avatarGroup-0-2-33 avatarGroup-d0-0-2-34"
            src="https://static.pro_boa.com/dev/upload/9f8f4041-acfe-4fcb-99b1-0545a6d35d1e.jpeg"
          />
        </div>
        <div
          className="textContainer-0-2-18"
        >
          <div>
            <p
              className="text-0-2-35 text-d0-0-2-36"
            >
              Richard Drissi
            </p>
            <p
              className="text-0-2-37 text-d0-0-2-38"
            >
               à crée l’équipe
            </p>
            <p
              className="text-0-2-39 text-d0-0-2-40"
            >
               Développement
            </p>
          </div>
          <p
            className="text-0-2-41 text-d0-0-2-42"
          >
            Il y a 20 min
          </p>
        </div>
      </div>
      <span
        className="icon-roundedUp icon-0-2-17"
      />
    </div>
    <div
      className="notificationCard-0-2-15 notificationCard-d2-0-2-43"
    >
      <div
        className="avatarContainer-0-2-16"
      >
        <div
          className="avatarContainer-0-2-21"
        >
          <div
            className="dot-0-2-20"
          />
          <img
            className="avatarGroup-0-2-44 avatarGroup-d0-0-2-45"
            src="https://static.pro_boa.com/dev/upload/9f8f4041-acfe-4fcb-99b1-0545a6d35d1e.jpeg"
          />
        </div>
        <div
          className="textContainer-0-2-18"
        >
          <div>
            <p
              className="text-0-2-46 text-d0-0-2-47"
            >
              Marouane Ait Mensour
            </p>
            <p
              className="text-0-2-48 text-d0-0-2-49"
            >
               à crée l’équipe
            </p>
            <p
              className="text-0-2-50 text-d0-0-2-51"
            >
               Project Manager
            </p>
          </div>
          <p
            className="text-0-2-52 text-d0-0-2-53"
          >
            Il y a 20 min
          </p>
        </div>
      </div>
      <span
        className="icon-roundedUp icon-0-2-17"
      />
    </div>
  </div>
  <div
    className="notificationFooter-0-2-5"
  >
    <p
      className="text-0-2-54 text-d0-0-2-55"
    >
      <div
        className="cursor-0-2-7"
        onClick={[Function]}
      >
        Afficher toutes les notifications
      </div>
    </p>
  </div>
</div>
`;
